modules/ipc/src/zenoh/topic_database.cpp

```diff
@@ -64,6 +70,33 @@ auto ZenohTopicDatabase::getTypeInfo(const std::string& topic) -> const serdes::
   return topics_type_db_[topic];
 }
 
+[[nodiscard]] auto ZenohTopicDatabase::getServiceTypeInfo(const std::string& topic)
```

Comment:
here as well,  I think there shouldn' t be an exception or checkfail if the retrieval fails. I have seen it fail before for publisher types. If we return an std::optional, the recepient can deal with the problem without try/catch. For us it even is not recoverable, because we disable exceptions -> this checkfails.

Reply:
yes, this is a very good point, I will update the behavior for both function in a follow up PR